pos_picking_delayed/models/pos_order.py

```diff
@@ -0,0 +1,30 @@
+# coding: utf-8
+# Copyright 2018 - Today Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import api, models
+
+
+class PosOrder(models.Model):
+    _inherit = 'pos.order'
+
+    @api.model
+    def create_from_ui(self, orders):
+        return super(PosOrder, self.with_context(
+            picking_creation_delayed=True)).create_from_ui(orders)
```

Comment:
Wouldn't it be interesting to add a configuration by PoS?

Reply:
Hi @legalsylvain , actually you don't use the option you set up picking_creation_delayed if I'm correct
Won't it be more readable to directly pass the context with the value of your option

Reply:
I guess it would be more logical and simple to check the pos config here and return the `super()` with the context flag or not.

Reply:
Not so easy. in create_from_ui, I don't have an easy access to the current pos_config, because it's an @api.model function, and ``orders`` is just a big json. I think parsing the json to know the session, and then the config, and then check the value of picking_creation_delayed will be more line of code.

The test is done later in ``create_picking`` function. (test of the context then test of the pos config setting).

Reply:
You're right there :slightly_smiling_face: Although you'd save splitting the recordset call in `create_picking`. I can't tell wich method would perfomance best, anyway :slightly_smiling_face:

Reply:
Yes, I split but I checked the code of create_picking and the first lines include a ``for order in self:``
So I don't understand how the code of this module could make the process slower. 

Réf : https://github.com/odoo/odoo/blob/10.0/addons/point_of_sale/models/pos_order.py#L643

don't you think ?

regards.

Reply:
Yes, my guess is that your option is the right one, therfore I approved :smile: